examples/applications/plot_time_series_lagged_features.py

```diff
@@ -0,0 +1,433 @@
+
+    for key, value in cv_results.items():
+        if key.startswith("test_"):
+            metric = key.split("test_")[1]
+            scores = consolidate_scores(cv_results, scores, metric)
+
+
+# %%
+# Modeling predictive uncertainty via quantile regression
+# -------------------------------------------------------
+# Instead of modeling the expected value of the distribution of
+# :math:`Y|X` like the least squares and Poisson losses do, one could try to
+# estimate quantiles of the conditional distribution.
+#
+# :math:`Y|X=x_i` is expected to be a random variable for a given data point
+# :math:`x_i` because we expect that the number of rentals cannot be 100%
+# accurately predicted from the features. It can be influenced by other
+# variables not properly captured by the existing lagged features. For
+# instance whether or not it will rain in the next hour cannot be fully
+# anticipated from the past hours bike rental data. This is what we
+# call aleatoric uncertainty.
+#
+# Quantile regression makes it possible to give a finer description of that
+# distribution without making strong assumptions on its shape.
+quantile_list = [0.05, 0.5, 0.95]
+
+for quantile in quantile_list:
+    model = HistGradientBoostingRegressor(loss="quantile", quantile=quantile)
+    cv_results = cross_validate(
+        model,
+        X,
+        y,
+        cv=ts_cv,
+        scoring=scoring,
+    )
+    time = cv_results["fit_time"]
+    scores["fit_time"].append(f"{time.mean():.2f} ± {time.std():.2f} s")
+
+    scores["loss"].append(f"quantile {int(quantile*100)}")
+    for key, value in cv_results.items():
+        if key.startswith("test_"):
+            metric = key.split("test_")[1]
+            scores = consolidate_scores(cv_results, scores, metric)
+
+df = pd.DataFrame(scores)
+
+styled_df_copy = df.copy()
+
+
+def extract_numeric(value):
+    parts = value.split("±")
+    mean_value = float(parts[0])
+    std_value = float(parts[1].split()[0])
+
+    return mean_value, std_value
+
+
+# Convert columns containing "±" to tuples of numerical values
+cols_to_convert = df.columns[1:]  # Exclude the "loss" column
+for col in cols_to_convert:
+    df[col] = df[col].apply(extract_numeric)
+
+min_values = df.min()
+
+# Create a mask for highlighting minimum values
+mask = pd.DataFrame("", index=df.index, columns=df.columns)
+for col in cols_to_convert:
+    mask[col] = df[col].apply(
+        lambda x: "font-weight: bold" if x == min_values[col] else ""
+    )
+
+styled_df_copy = styled_df_copy.style.apply(lambda x: mask, axis=None)
+styled_df_copy
+
+
+# %%
+# The minimal value for the Mean Pinball Loss is achieved by the GBRT
+# with the corresponding quantile. Similarly, the lowest RMSE is obtained
+# by the "gbrt_mse" model, as expected. A bit less intuitive, the best MAPE
+# corresponds to the "gbrt_poisson" model. This is due to the local nature of
+# the Poisson Loss normalization.

```

Comment:
At some point we got rid of those names for the models, so let's rephrase and be more accurate on the analysis:
```suggestion
# Even if the score distributions overlap due to the variance in the dataset, it
# is true that the average RMSE is lower when `loss="squared_error"`, whereas
# the average MAPE is lower when `loss="absolute_error"` as expected. That is
# also the case for the Mean Pinball Loss with the quantiles 5 and 95. The score
# corresponding to the 50 quantile loss is overlapping with the score obtained
# by minimizing other loss functions, which is also the case for the MAE.
```